### ISSUE TITLE
update-resolv-conf outdated/openresolv necessary?

### ISSUE THREAD

**Issue by bspinner** [haugene/docker-transmission-openvpn#229]
Hi, 
as I've mentioned in #225 , I had to make some modifications, so update-resolv-conf gets called.

While doing that, I've noticed that `update-resolv-conf` seems to depending on `openresolv`, which isn't installed by default in this image. Is `resolvconf` just misbehaving for me or is `openresolv` a missing dependency indeed?

Second, I wondered whether the update-resolv-conf script comes from [this](https://github.com/masterkorp/openvpn-update-resolv-conf) repo. Is that the case? If so, were modifications made or is the script in the mentioned repo just more up to date and should be merged into this repo?

**Comment by haugene** [comment]
Hi,

Had a look at this now. And you're right that `openresolv` is missing in the image. As for the `update-resolv-conf`script, that is added during the installation of openvpn and not modified from this project.

The config files for PIA does not reference running of `update-resolv-conf` and the question then becomes if we should add your changes to the image so that they're run by all providers or make it possible to customize this using run parameters.

Would it be a possible solution to add installation of  `openresolv` to this image and making it possible to override the TRANSMISSION_CONTROL_OPTS set in openvpn/start.sh? Then you could start the container with something like `TRANSMISSION_CONTROL_OPTS=--up '/etc/openvpn/update-resolv-conf && /etc/transmission/start.sh'`?

**Comment by bspinner** [comment]
> Would it be a possible solution to add installation of openresolv to this image and making it possible to override the `TRANSMISSION_CONTROL_OPTS` set in `openvpn/start.sh`? Then you could start the container with something like `TRANSMISSION_CONTROL_OPTS=--up '/etc/openvpn/update-resolv-conf && /etc/transmission/start.sh'`?

That should do it.

**Comment by haugene** [comment]
@h45rd. Thoughts? The oldest open issue. Not sure what to do with it.
Are you still using this container @bspinner? Any input two years later? ;)

**Comment by h45rd** [comment]
Hmm.. not sure what would be best on a global scale.
Personally, I prefer setting a third party DNS myself to leaving DNS resolution up to the VPN provider. Both from a security and a "container maintainability" standpoint.

During the search for a possible new VPN provider in the last couple of days, I tested several with this container.
And man let me tell you.. they all do their own thing when it comes to pushing options to the clients. Some push nothing, others push a plethora of usless stuff, like IPv6 routes and gateways that are not working, etc.
I ended up filtering most pushes by default to get things working the way I want them to without having to do a lot of modifications for each provider switch.

Thought that was a bit silly before.. but now I also understand why people are coming back to this container, even when they do not need transmission: everything is tried, tested and works out of the box. It is actively maintained and up to date and the community is lively and active. And there are some very nifty features and fine-tuning possibilities for those who need them.
Try finding a simple OpenVPN container you could say the same about..


The oldest open issue warrants a bit of OT for nostalgia's sake:
Hehe. Man.. you have been quite busy in the repo the last couple of days :+1: 

I sometimes forget how popular this container is/has become. Had to google something concerning OpenVPN earlier today and had a lot of hits from this little "forum" :)
Sidebar: the other day I was reading an old computer magazine at the dentist and they actually recommended this container for secure torrenting. Crazy stuff :D

Your project was one of the first things I tried when I started out with docker. Following it's development and the issues has taught me a lot. Thanks for that :)

**Comment by haugene** [comment]
Hah, yeah. I got down some hours last Sunday - punching through all the issues. Felt good to do a bit of spring cleaning. And then I took a week off ;)

A bit of OT is appreciated 👍 Thanks for sharing, pretty cool that it's in a magazine as well 😮One thing is tutorials online, but in print as well - who would have thought. This was my first Docker project as well, and I've learnt a lot along the way. I'm glad to hear you've learnt from it too.

But - to this issue. Then I think I'll just close it. We could consider adding dns flags to the "default" recommended run command in the README. And then the official way of running it is to specify it manually.

For anyone reading this thread later, feel free to open a PR installing update-resolv-conf or somehow make it a built in optional feature.
